File: test-upload.js
import FormData from 'form-data';
import fs from 'fs';
import fetch from 'node-fetch';

async function testBatchUpload() {
  console.log('Testing batch upload with proper flow...\n');
  
  // Step 1: Upload and preview file
  const form1 = new FormData();
  const csvContent = 'payee_name\nMicrosoft Corporation\nApple Inc\nAmazon.com Inc';
  form1.append('file', Buffer.from(csvContent), 'test-batch.csv');
  
  console.log('1. Uploading file for preview...');
  const previewResponse = await fetch('http://localhost:5000/api/upload/preview', {
    method: 'POST',
    body: form1,
    headers: form1.getHeaders()
  });
  
  if (!previewResponse.ok) {
    console.error('Preview failed:', await previewResponse.text());
    return;
  }
  
  const previewData = await previewResponse.json();
  console.log('✅ Preview successful');
  console.log('   Headers:', previewData.headers);
  console.log('   Temp file:', previewData.tempFileName);
  
  // Step 2: Process the file
  console.log('\n2. Processing file...');
  const processResponse = await fetch('http://localhost:5000/api/upload/process', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      tempFileName: previewData.tempFileName,
      originalFilename: previewData.filename,
      payeeColumn: 'payee_name',
      matchingOptions: {
        enableFinexio: true,
        enableMastercard: false,
        enableAddressValidation: false,
        enableAkkio: false
      }
    })
  });
  
  if (!processResponse.ok) {
    console.error('Process failed:', await processResponse.text());
    return;
  }
  
  const processData = await processResponse.json();
  console.log('✅ Batch created successfully');
  console.log('   Batch ID:', processData.batchId);
  
  // Step 3: Check status
  console.log('\n3. Checking batch status...');
  const statusResponse = await fetch(`http://localhost:5000/api/upload/batches/${processData.batchId}`);
  const statusData = await statusResponse.json();
  console.log('   Status:', statusData.status);
  console.log('   Records:', `${statusData.processedRecords}/${statusData.totalRecords}`);
  
  console.log('\n✅ Batch upload test completed successfully!');
}

testBatchUpload().catch(console.error);

File: test-upload.py
#!/usr/bin/env python3
import requests
import json
import time

# Read the test CSV file
with open('test-enrichment-flow.csv', 'rb') as f:
    files = {'file': ('test-enrichment-flow.csv', f, 'text/csv')}
    data = {
        'enableFinexio': 'true',
        'enableMastercard': 'true',
        'enableGoogleAddress': 'true',
        'enableAkkio': 'true'
    }
    
    print("📤 Uploading test file...")
    response = requests.post('http://localhost:5000/api/upload', files=files, data=data)
    
    if response.status_code == 200:
        result = response.json()
        batch_id = result.get('batchId')
        print(f"✅ Upload successful! Batch ID: {batch_id}")
        print(f"📊 Response: {json.dumps(result, indent=2)}")
        
        # Monitor the batch status
        print("\n📍 Monitoring batch processing...")
        for i in range(30):  # Check for up to 5 minutes
            time.sleep(10)  # Check every 10 seconds
            
            status_response = requests.get(f'http://localhost:5000/api/upload/batches')
            if status_response.status_code == 200:
                batches = status_response.json()
                batch = next((b for b in batches if b['id'] == batch_id), None)
                
                if batch:
                    print(f"\n⏱️  Time: {i*10}s")
                    print(f"📊 Status: {batch['status']}")
                    print(f"📝 Current Step: {batch.get('currentStep', 'N/A')}")
                    print(f"📊 Classification: {batch.get('processedRecords', 0)}/{batch.get('totalRecords', 0)}")
                    print(f"🔍 Finexio: {batch.get('finexioMatchingStatus', 'pending')}")
                    print(f"📍 Google Address: {batch.get('googleAddressStatus', 'pending')}")
                    print(f"💳 Mastercard: {batch.get('mastercardEnrichmentStatus', 'pending')}")
                    print(f"🤖 Akkio: {batch.get('akkioPredictionStatus', 'pending')}")
                    
                    if batch['status'] == 'completed':
                        print("\n✅ Batch processing completed!")
                        break
            else:
                print(f"❌ Failed to get batch status: {status_response.status_code}")
    else:
        print(f"❌ Upload failed with status code: {response.status_code}")
        print(f"Response: {response.text}")